presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/thrift/ThriftMetastoreUtil.java

```diff
@@ -497,6 +497,11 @@ public static long getLastDataCommitTime(org.apache.hadoop.hive.metastore.api.Pa
 
     public static Partition fromMetastoreApiPartition(org.apache.hadoop.hive.metastore.api.Partition partition, PartitionMutator partitionMutator, ColumnConverter columnConverter)
     {
+        if (partition instanceof Partition) {
```

Comment:
Wondering if we should do this at the caller itself, basically avoid calling fromMetastoreApiPartition ( to transform using partitionMutator and columnConverter) if the partition is already instance of Partition. Also do we loose the sd validation that is happening at line#505 if we return early.

Reply:
> Wondering if we should do this at the caller itself, basically avoid calling fromMetastoreApiPartition ( to transform using partitionMutator and columnConverter) if the partition is already instance of Partition.

I want the caller code to be clean and not worry about the object returned and let the util method handle it.

> Also do we loose the sd validation that is happening at line#505 if we return early?

The null check ? That is done when the Partition object is created right. Like the Partition does not take in a null sd/ Storage.